bowling/score.js

```diff
@@ -0,0 +1,79 @@
+const Frame = require("./frame")
+
+class Scoreboard {
+  constructor() {
+    this.frames = []
+    
+  }
+
+  addFrame(...rolls){
+    return this.frames.push(
+      new Frame(rolls)
+    )
+  }
+
+  calculateScore() {
+    let score = 0
+    //strike
+    for (let i = 0; this.frames.length > i; i++) {
+      let currentFrame = this.frames[i]
+      if(currentFrame.isStrike() && this.hasNext(i)){
+        let nextFrame = this.frames[i + 1]
+        let spareBonus = 0
+       if (nextFrame.isStrike()) {
+          spareBonus = nextFrame.rolls[0]
+        } else {
+          spareBonus = nextFrame.rolls[0] + nextFrame.rolls[1]
+        }
+
+        //for simplicity, as bonus is always the same as the first rolls,
+        // Im not giving bonus for the 10th only the points in the first rolls 
+        if(!this.is10th(i)){
+          nextFrame.addBonus(spareBonus) 
+        }
+
+
+      } //spare
+      else if (currentFrame.isSpare() && this.hasNext(i)) {
```

Comment:
Consider breaking this out into a separate function as this function is a little long